hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,749 @@
+            and ignored if data is a Dataset
+        """
+        self._initialize_grid()
+        # NOTE: variables in a dataset are not longer renamed as used to be the case in
+        # set_staticmaps
+        name_required = isinstance(data, np.ndarray) or (
+            isinstance(data, xr.DataArray) and data.name is None
+        )
+        if name is None and name_required:
+            raise ValueError(f"Unable to set {type(data).__name__} data without a name")
+        if isinstance(data, np.ndarray):
+            if data.shape != self._data.raster.shape:
+                raise ValueError("Shape of data and grid maps do not match")
+            data = xr.DataArray(dims=self._data.raster.dims, data=data, name=name)
+        elif isinstance(data, xr.DataArray) and name_required:
+            data.name = name
+            data = data.to_dataset()
+        elif not isinstance(data, xr.Dataset):
+            raise ValueError(f"cannot set data of type {type(data).__name__}")
+
+        if len(self._data) == 0:  # empty grid
+            self._data = data
+        else:
+            for dvar in data.data_vars:
+                if dvar in self._data:
+                    if read:
+                        self.logger.warning(f"Replacing grid map: {dvar}")
+                self._data[dvar] = data[dvar]
+
+    def write(
+        self,
+        fn: str = "grid/grid.nc",
+        gdal_compliant: bool = False,
+        rename_dims: bool = False,
+        force_sn: bool = False,
+        write: bool = False,
+        **kwargs,
+    ) -> DeferedFileClose | None:
+        """Write model grid data to netcdf file at <root>/<fn>.
+
+        key-word arguments are passed to :py:meth:`~hydromt.models.Model.write_nc`
+
+        Parameters
+        ----------
+        fn : str, optional
+            filename relative to model root, by default 'grid/grid.nc'
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `write_nc` method.
+        gdal_compliant : bool, optional
+            If True, write grid data in a way that is compatible with GDAL,
+            by default False
+        rename_dims: bool, optional
+            If True and gdal_compliant, rename x_dim and y_dim to standard names
+            depending on the CRS (x/y for projected and lat/lon for geographic).
+        force_sn: bool, optional
+            If True and gdal_compliant, forces the dataset to have
+            South -> North orientation.
+        """
+        if len(self._data) == 0:
+            _exec_nodata_strat(
+                msg="No grid data found, skip writing.",
+                strategy=NoDataStrategy.IGNORE,
+                logger=logger,
+            )
+        else:
+            _assert_write_mode(write)
+            # write_nc requires dict - use dummy 'grid' key
+            return write_nc(
+                {"grid": self._data},  # TODO: use a name argument or fn as key
```

Comment:
Not needed to use a name argument or fn as key. The "grid" name is only temporary used for the write_nc function, is not appearing anywhere in the output file name or properties and disappears as soon as write_nc is done running :)